docs/guides/migrations/guide.rst

```diff
@@ -0,0 +1,1298 @@
+
+Unseen to the user, EdgeDB Watch will begin creating individual migration 
+scripts for every time you save a change to one of your files. These 
+are stored as separate "dev mode" migrations, which are sort of like 
+preliminary migrations that haven't been turned into a standalone 
+migration script yet.
+
+If you start with this schema:
+
+.. code-block:: sdl
+
+    module default {
+      type User {
+        name: str;
+      }
+    }
+
+And then add a single property:
+
+.. code-block:: sdl
+
+    module default {
+      type User {
+        name: str;
+        number: int32;
+      }
+    }
+
+You will see EdgeDB Watch quickly display "calculating diff" before 
+disappearing once the difference has been calculated and deemed to 
+be good.
+
+However, if you add incorrect syntax to the schema:
+
+.. code-block:: sdl
+
+    module default {
+      type User {
+        name: str;
+        number: int32;
+        wrong_property: i32; # Should say int32, not i32
+      }
+    }
+
+Then EdgeDB Watch will suddenly pipe up and inform you that the schema 
+can't be resolved:
+
+.. code-block::
+
+    error: type 'default::i32' does not exist
+    ┌─ myproject/dbschema/default.esdl:5:25
+    │
+    5 │         wrong_property: i32;
+    │                         ^^^ error
+
+    Schema migration error:
+    cannot proceed until .esdl files are fixed
+
+Once you correct the ``i32`` property to ``int32``, EdgeDB Watch will 
+let you know that things are okay now, and will become quiet again:
+
+.. code-block::
+
+    Resolved. Schema is up to date now.
+
+EdgeDB Watch is best run in a separate tab on your command line so 
+that you can take care of other tasks—including officially migrating 
+when you are satisfied with your current schema—without having to 
+stop the process.
+
+If you are curious what is happening as EdgeDB Watch does its thing, 
+try the following query after you have made some changes:
+
+.. code-block::
+
+    group schema::Migration {
+        name,
+        script
+    } by .generated_by;
+
+Some migrations will contain nothing in their ``generated_by`` property, 
+while those generated by EdgeDB Watch will have a 
+``MigrationGeneratedBy.DevMode``. 
+
+(The final option for ``generated_by`` is
+``MigrationGeneratedBy.DDLStatement``, which will show up if you directly
+change your schema by using DDL - which is generally not recommended)
+
+Once you are satisfied with your changes while running EdgeDB Watch, 
+just create the migration with ``edgedb migration create`` and then 
+apply them with one small change: ``edgedb migrate --dev-mode`` to 
+let the CLI know to apply the migrations made during dev mode that 
+were made by EdgeDB Watch.
+
+So you really want to use DDL
+=============================
+
+You might have a good reason to use a direct DDL statement or two 
+to change your schema. How do you make that happen? EdgeDB disables 
+the usage of DDL by default, so this attempt to use DDL will not work:
```

Comment:
DDL is only disabled after you run your first migration, right? If so, we should note that.

Reply:
Interesting! Did not know that. https://github.com/edgedb/edgedb/pull/6656/commits/f16ce487aae31069feda0574ec7da1af8aa117fa